resources/js/features/authentication/forms/login/Login.form.tsx

```diff
@@ -0,0 +1,55 @@
+import { FormConfigWithoutTransformFn } from "@felte/core";
+import { useTransContext } from "@mbarzda/solid-i18next";
+import { createMutation, useQueryClient } from "@tanstack/solid-query";
+import { FelteForm, FelteSubmit } from "components/felte-form";
+import { TextField } from "components/ui";
+import { User } from "data-access/memo-api";
+import { Component } from "solid-js";
+import { z } from "zod";
+
+export namespace LoginForm {
+  export const schema = z.object({
+    email: z.string(),
+    password: z.string(),
+  });
+
+  export const initialValues: Input = {
+    email: "",
+    password: "",
+  };
+
+  export type Input = z.input<typeof schema>;
+  export type Output = z.output<typeof schema>;
+
+  export const Component: Component = () => {
+    const [t] = useTransContext();
+    const queryClient = useQueryClient();
+    const mutation = createMutation(() => ({
+      mutationFn: User.login,
+      onSuccess() {
+        queryClient.invalidateQueries({ queryKey: User.keys.status() });
+      },
+    }));
+
+    const onSubmit: FormConfigWithoutTransformFn<LoginForm.Output>["onSubmit"] =
+      async (values) => {
+        await mutation.mutateAsync({
+          ...values,
```

Comment:
Jak rozumiem to działa dlatego że schema tu wyżej ma strukturę pasującą do LoginRequest, tak? Jeśli tak, dodaj może komentarz przy schema który mówi że to ważne. Że na przykład jak zmienisz w tym pliku wszystkie password na pass to rozwalisz aplikację. Bo teraz to nie jest za bardzo widoczne.

A w ogóle, mało wierzę że defensywna kopia jest tutaj potrzebna.

Reply:
kopia może niepotrzebna. formularz logowania to akurat łatwy przypadek. Przy tworzeniu duzych formularzy lepiej będzie mieć dedykowane funkcje do parsowania stanu formularza na payload backendowy. Wtedy można ładnie otypować i mieć pewność, że jak model się zmieni, to trzeba poprawić funkcje, które z niego korzystają

Reply:
chyba doszliśmy do konsensusu jakiegoś

Reply:
W pierwszym komentarzu pisałem że niepokoi mnie, że ta schema musi być dopasowana do LoginRequest a kompilator tego nie pilnuje. Ale nie miałem racji bo jednak kompilator pilnuje dzięki tym generycznym trikom, więc jest git.

Natomiast nadal jestem przeciwny robieniu kopii gdy jest niepotrzebna, bo wygląda to jakby była potrzebna i powoduje FUD, przynajmniej u mnie.

Reply:
usunąłem te kopię